app/models/comment.rb

```diff
@@ -0,0 +1,12 @@
+class Comment < ApplicationRecord
+  belongs_to :post, foreign_key: true, optional: true
+
+  def update_post_commentcount
+    post = Post.find(post_id)
+    _
+    post.comments_count = 0 if post.comments_count.nil?
+
+    post.comments_count += 1
+    post.save
+  end
```

Comment:
- [ OPTIONAL] You could also use the Callback `after_save` here as proposed for the posts counts